src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs

```diff
@@ -168,6 +177,51 @@ private static ScriptHostConfiguration GetScriptHostConfiguration(string scriptP
             return scriptHostConfig;
         }
 
+        private static string MakeValidHostId(string id)
```

Comment:
Also, you should consider the implications of deriving an ID from the current directory path. The ID for a function app (set of functions) is supposed to be a static identifier that doesn't change. Logging and other storage artifacts depend on that. E.g. when set explicitly in host.json that ID will travel around with the files. In your case it will change if the app is copied to a different directory, machine, etc.